src/client.js

```diff
@@ -305,427 +305,558 @@ class ConnectionParameters {
     }
 
-}
+  }
 
-// ========================================================================================
-// ACC Client
-// ========================================================================================
+  // ========================================================================================
+  // File Uploader
+  // ========================================================================================
+
+  const fileUploader = (client) => {
+    /**
+         * Will call the SOAP method(IncreaseValue) to increament the counter
+         * @returns {Promise<number>}
+         */
+    const _increaseValue = async () => {
+      const xtkCounter= await client.NLWS.xtkCounter.increaseValue({name: 'xtkResource'});
+      return xtkCounter
+    }
+
+    /**
+         * This will create and return fileRes object
+         * @param counter
+         * @param data
+         * @returns {{originalName, internalName: string, fileName, useMd5AsFilename: string, xtkschema: string, storageType: number, label, md5: (string|string|string|null|any|number)}}
+         * @private
+         */
+    const _createFileRes = (counter, data) => {
+      return {
+        internalName: 'RES' + counter,
+        md5: data[0].md5,
+        label: data[0].fileName,
+        fileName: data[0].fileName,
+        originalName: data[0].fileName,
+        useMd5AsFilename: '1',
+        storageType: 5,
+        xtkschema: 'xtk:fileRes'
+
+      }
+    }
+
+    /**
+         * This will write fileRes object in the system.
+         * @param fileRes
+         * @returns {Promise<void>}
+         * @private
+         */
+    const _write = async (fileRes) => {
+      await client.NLWS.xtkSession.write(fileRes);
+    }
 
-/**
+    /**
+         * This will call the SOAP method(PublishIfNeeded) to publish the fileRes object
+         * @param fileRes
+         * @returns {Promise<void>}
+         * @private
+         */
+    const _publishIfNeeded = async (fileRes) => {
+      await client.NLWS.xtkFileRes.create(fileRes).publishIfNeeded();
+    }
+
+    /**
+         * This Will call the SOAP method(GetURL) which returns the public URL of the uploaded file.
+         * @param fileRes
+         * @returns {Promise<string>}
+         * @private
+         */
+    const _getPublicUrl = async (fileRes) => {
+      return await client.NLWS.xtkFileRes.create(fileRes).getURL()
+    }
+    return {
+      /**
+             * This is the exposed/public method for fileUploader instance which will do all the processing related to the upload process internally and returns the promise containing all the required data.
+             * @param file
+             * @returns {Promise<{name: string, md5: string, type: string, size: string, url: string}>}
+             */
+      upload: (file) => {
+        return new Promise((resolve, reject) => {
+          if (!Util.isBrowser()) {
+            reject('File uploading is only supported in browser based calls.');
```

Comment:
All exception types must be consistent (see CampaignException)

Reply:
Done

Reply:
I don't believe this will return the right kind of exception: calling reject here will not be caught by the catch blocks below